src/OpenTelemetry/Internal/Base2ExponentialHistogramHelper.cs

```diff
@@ -0,0 +1,120 @@
+// <copyright file="Base2ExponentialHistogramHelper.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Internal;
+
+internal static class Base2ExponentialHistogramHelper
+{
+    private const double EpsilonTimes2 = double.Epsilon * 2;
+    private static readonly double Ln2 = Math.Log(2);
+
+    public static double LowerBoundary(int index, int scale)
+    {
+        if (scale > 0)
+        {
+#if NET6_0_OR_GREATER
+            var inverseFactor = Math.ScaleB(Ln2, -scale);
+#else
+            var inverseFactor = ScaleB(Ln2, -scale);
+#endif
+            var lowerBound = Math.Exp(index * inverseFactor);
+            return lowerBound == 0 ? double.Epsilon : lowerBound;
+        }
+        else
+        {
+            if ((scale == -1 && index == -537) || (scale == 0 && index == -1074))
+            {
+                return EpsilonTimes2;
+            }
+
+            var n = index << -scale;
+
+            // LowerBoundary should not return zero.
+            // It should return values >= double.Epsilon (2 ^ -1074).
+            // n < -1074 occurs at the minimum index of a scale.
+            // e.g., At scale -1, minimum index is -538. -538 << 1 = -1075
+            if (n < -1074)
+            {
+                return double.Epsilon;
+            }
+
+#if NET6_0_OR_GREATER
+            return Math.ScaleB(1, n);
```

Comment:
Left-shift is probably faster?
```suggestion
            return 2 << (n - 1);
```

Reply:
Math.ScaleB(1, n) != 2 << (n - 1) for n < 1. Some math involving a right shift might work? I don't think this is a super perf critical path.

Reply:
Got it.